be/src/vec/data_types/data_type_decimal.h

```diff
@@ -206,6 +206,32 @@ class DataTypeDecimal final : public IDataType {
     void to_pb_column_meta(PColumnMeta* col_meta) const override;
 
     Field get_default() const override;
+
+    Field get_field(const TExprNode& node) const override {
+        DCHECK_EQ(node.node_type, TExprNodeType::DECIMAL_LITERAL);
+        DCHECK(node.__isset.decimal_literal);
+        // decimalv2
+        if constexpr (std::is_same_v<TypeId<T>, TypeId<Decimal128>>) {
```

Comment:
do not use is_same_v to check if it is decimalv2.
data type ptr has a method TypeIndex get_type_id() const override { return TypeId<T>::value; } it will return the type clearly.